server/middleware.go

```diff
@@ -147,6 +148,81 @@ func requireApp(appRepo datastore.ApplicationRepository) func(next http.Handler)
 	}
 }
 
+func requireAppPortalApplication(appRepo datastore.ApplicationRepository) func(next http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+
+			appID := chi.URLParam(r, "appID")
+
+			if util.IsStringEmpty(appID) {
+				appID = r.URL.Query().Get("appId")
+			}
+
+			app, err := appRepo.FindApplicationByID(r.Context(), appID)
+			if err != nil {
+
+				event := "an error occurred while retrieving app details"
+				statusCode := http.StatusBadRequest
+
+				if errors.Is(err, datastore.ErrApplicationNotFound) {
+					event = err.Error()
+					statusCode = http.StatusBadRequest
+				}
+
+				_ = render.Render(w, r, newErrorResponse(event, statusCode))
+				return
+			}
+
+			r = r.WithContext(setApplicationInContext(r.Context(), app))
+			next.ServeHTTP(w, r)
+		})
+	}
+}
+
+func requireAppPortalPermission(role auth.RoleType) func(next http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			authUser := getAuthUserFromContext(r.Context())
+			if authUser.Role.Type.Is(auth.RoleSuperUser) {
+				// superuser has access to everything
+				next.ServeHTTP(w, r)
+				return
+			}
+
+			if !authUser.Role.Type.Is(role) {
+				_ = render.Render(w, r, newErrorResponse("unauthorized role", http.StatusUnauthorized))
+				return
+			}
+
+			group := getGroupFromContext(r.Context())
+			for _, v := range authUser.Role.Groups {
+				if group.Name == v || group.UID == v {
```

Comment:
@Dotunj Since we already decided that applications should be unique is there any point in doing a group check again?